sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -729,6 +729,15 @@ object functions {
     }
   }
 
+  /**
+   * Aggregate function: returns the product of all numerical elements in a group.
+   *
+   * @group agg_funcs
+   * @since 3.2.0
+   */
+  def product(e: Column): Column =
+    withAggregateFunction { new Product(e.expr) }
```

Comment:
Don't we want to expose `def product(columnName: String): Column`?
Other agg funcs seem to have the overloaded functions.

Reply:
ah, new functions won't expose the signature.